sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/templating/JavaParserTemplateProcessor.java

```diff
@@ -310,13 +313,49 @@ private void configureInternalMethod(MethodDeclaration internalMethod, HttpReque
         BlockStmt body = internalMethod.getBody().get();
 
         initializeHttpRequest(body, method);
-        boolean serializationFormatSet = addRequestBody(body, method);
         addRequestOptionsToRequestIfPresent(body, method);
+        boolean serializationFormatSet = addRequestBody(body, method);
+        addStaticHeaders(body, method);
         finalizeHttpRequest(body, method.getMethodReturnType(), method, serializationFormatSet);
-
         internalMethod.setBody(body);
     }
 
+    void addStaticHeaders(BlockStmt body, HttpRequestContext method) {
+        // Headers from HttpRequestInformation always take precedence over inferred headers from body types
+        if (!CoreUtils.isNullOrEmpty(method.getHeaders())) {
+            String[] requestHeaders = method.getHeaders();
+            StringBuilder statementBuilder = new StringBuilder("httpRequest.getHeaders()");
+            for (String requestHeader : requestHeaders) {
+                int colonIndex = requestHeader.indexOf(":");
```

Comment:
Few minor high-level thoughts:

1. Should we support a header value that's just `headerName`, possibly with null as the header value?
2. Or, in cases that we don't want to support, such as point 1, should we throw an error here?

Reply:
I don't think there's a usecase for sending a null header value with only header name. So, we should drop that header or throw.

Reply:
yep, updated it to throw.